processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
> In your mapper method, you have Car toCar(CarDto s) and void intoCar(CarDto s, @MappingTarget Car t) - if the current logic of auto-inheritance was applied, but would include the methods in the current mapper as well...

True, that case also came to my mind as a problem.

Ok, so I understood the AUTO mode to be more general than you guys had it in mind apparently. Given that I'm not big fan of it altogether I don't mind to limit its scope :) I feel AUTO is indeed a too generic name in this case, though (sorry for the confusion, as said, I understood you'd mean to auto-inherit *all* matching mappings). Maybe AUTO_INHERIT_FROM_CONFIG? That'd leave the door open for other flavors of auto-configuration down the road if needed.